File: 卡码/代码随想录/1/01-数组/Go/01-二分/LeetCode_0035.go
//35. 搜索插入位置
package main

import "fmt"

func searchInsert(nums []int, target int) int {
	left,right:=0,len(nums)-1
	for left<=right {
		middle:=(left+right)/2
		if nums[middle]==target {
			return middle
		}else if target<nums[middle] {
			right=middle-1
		}else{
			left=middle+1
		}
	}
	return left
}

func main(){
	nums := []int{1,3,5,6}
	fmt.Println(searchInsert(nums,5))
	fmt.Println(searchInsert(nums,2))
	fmt.Println(searchInsert(nums,7))
}

File: 简单/Java/easy/src/main/java/com/kurumi/leetCode/easy/二分查找/LeetCode_0035.java
package com.kurumi.leetCode.easy.二分查找;

/**
 * @Author Administrator
 * @Date 2026年6月13日 19:13:08
 * @Description 35. 搜索插入位置
 */
public class LeetCode_0035 {
    public int searchInsert(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return left;
    }
}
